walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -15,9 +15,136 @@
+                            eventsIterator.remove();
+                        }      
+                        // If there are only optional attendees & no times are available for them
+                        if (!attendeeEntry.getValue() && availableTimes.isEmpty()) {
+                            availableTimes.add(TimeRange.WHOLE_DAY);
+                        }
+                    }
+                } 
+            }
+        } 
+        return availableTimes;
+    }
+
+    public void blockTimeRange(TimeRange blockedTimeRange, long meetingDuration, boolean mandatory) {
+        for (int i = 0; i < availableTimes.size(); i++) {
+            TimeRange timeRange = availableTimes.get(i);
+
+            if ((blockedTimeRange.equals(TimeRange.WHOLE_DAY) && !mandatory) 
+                || blockedTimeRange.duration() < meetingDuration) return;
+
+            if (timeRange.overlaps(blockedTimeRange)) {
+
+                TimeRange newTimeRange1 = TimeRange.fromStartEnd(timeRange.start(), 
+                                                blockedTimeRange.start(), false);
+                TimeRange newTimeRange2 = TimeRange.fromStartEnd(blockedTimeRange.end(), 
+                                                timeRange.end(), false);      
+                /*
+                            |--blocked--|
+                        |-----timeRange----|
+                */
+                if (timeRange.contains(blockedTimeRange)) {
+                    System.out.println(timeRange.toString() + " contains " + blockedTimeRange.toString());
+                    if (blockedTimeRange.start() != timeRange.start()) {
+                        createNewTimeRange(i, meetingDuration, newTimeRange1);
+                    }
+                    if (blockedTimeRange.end() != timeRange.end()) {
+                        createNewTimeRange(i + 1, meetingDuration, newTimeRange2);
+                    }        
+                }
+                /*
+                            |--blocked--|
+                    |--timeRange--|
+                */
+                else if (blockedTimeRange.end() > timeRange.end()) {
+                    System.out.println(blockedTimeRange.toString() + " overlaps end of " + timeRange.toString());
+                    createNewTimeRange(i, meetingDuration, newTimeRange1);    
+                }
+                /*
+                    |--blocked--|
+                          |--timeRange--|
+                */
+                else if (blockedTimeRange.start() < timeRange.start()) {
+                    System.out.println(blockedTimeRange.toString() + " overlaps start of " + timeRange.toString());
+                    createNewTimeRange(i, meetingDuration, newTimeRange2);              
+                }
+                availableTimes.remove(timeRange);
+                return;
+            }
+        }
+    }
+
+    public void createNewTimeRange(int index, long meetingDuration, TimeRange timeRange) {
```

Comment:
Let's think about the naming and/or the function of this method a big more, since you're not really creating any time ranges here.